### PR TITLE
Remove default validator from nats tests

### PR DESCRIPTION
**Pull request by FriedrichWilken**
**Description**
In `components/eventing-controller/controllers/subscription/nats/reconciler_test.go` the [startReconciler](https://github.com/kyma-project/kyma/blob/feceb796ee1dc250fa2280a6bdf806358d65fbd8/components/eventing-controller/controllers/subscription/nats/reconciler_test.go#L799) function requires a `sinkValidator`:

```go
func startReconciler(eventTypePrefix string, sinkValidator sinkValidator, natsURL string) context.CancelFunc {
```

it further [creates a new reconciler](https://github.com/kyma-project/kyma/blob/feceb796ee1dc250fa2280a6bdf806358d65fbd8/components/eventing-controller/controllers/subscription/nats/reconciler_test.go#L828) and then [sets](https://github.com/kyma-project/kyma/blob/feceb796ee1dc250fa2280a6bdf806358d65fbd8/components/eventing-controller/controllers/subscription/nats/reconciler_test.go#L838) the `reconciler`'s `sinkValidator` to the aforementioned `sinkValidator` that was passed as an parameter.

```go
	reconciler = NewReconciler(
		ctx,
		k8sManager.GetClient(),
		applicationLister,
		k8sManager.GetCache(),
		defaultLogger,
		k8sManager.GetEventRecorderFor("eventing-controller-nats"),
		envConf,
		defaultSubsConfig,
	)
	reconciler.sinkValidator = sinkValidator
```

All the tests that [use](https://github.com/kyma-project/kyma/blob/feceb796ee1dc250fa2280a6bdf806358d65fbd8/components/eventing-controller/controllers/subscription/nats/reconciler_test.go#L84) `startReconciler` only pass the [default](https://github.com/kyma-project/kyma/blob/feceb796ee1dc250fa2280a6bdf806358d65fbd8/components/eventing-controller/controllers/subscription/nats/reconciler.go#L351) validator (`defaultSinkValidator`).

Thus, passing a validator is unnecessary because `NewReconciler` already [sets](https://github.com/kyma-project/kyma/blob/feceb796ee1dc250fa2280a6bdf806358d65fbd8/components/eventing-controller/controllers/subscription/nats/reconciler.go#L68) the `sinkValidator` to [the very same](https://github.com/kyma-project/kyma/blob/feceb796ee1dc250fa2280a6bdf806358d65fbd8/components/eventing-controller/controllers/subscription/nats/reconciler.go#L77) `defaultSinkValidator`:
```go
reconciler := &Reconciler{
		ctx:                 ctx,
		Client:              client,
		Cache:               cache,
		logger:              logger,
		recorder:            recorder,
		eventTypeCleaner:    eventtype.NewCleaner(cfg.EventTypePrefix, applicationLister, logger),
		sinkValidator:       defaultSinkValidator,
		customEventsChannel: make(chan event.GenericEvent),
	}
```

This PR removes the `sinkValidator` parameter from `startReconciler`.

Furthermore there are minor refactorings to functions to create subscriptions in tests and the usages of these functions.

This is a precursor to further refactoring.

### CODE REVIEW

File: components/eventing-controller/testing/test_helpers.go [hunk 1]
```diff
@@ -329,11 +333,24 @@ func WithEventTypeFilter(s *eventingv1alpha1.Subscription) {
...
-func WithValidSink(svcNs, svcName string, s *eventingv1alpha1.Subscription) {
-	s.Spec.Sink = GetValidSink(svcNs, svcName)
+// WithValidSink is SubscriptionOpt for creating a subscription with a valid sink that itself gets created from
+// the svcNamespace and the svcName.
+func WithValidSink(svcNamespace, svcName string) SubscriptionOpt {
+	return func(s *eventingv1alpha1.Subscription) { s.Spec.Sink = NewValidSink(svcNamespace, svcName) }
 }
 
-func GetValidSink(svcNs, svcName string) string {
+// WithSpecificSink is SubscriptionOpt for creating a subscription with a specific sink.
```
Comment (FriedrichWilken) [inline components/eventing-controller/testing/test_helpers.go:342]:
```suggestion
// WithSpecificSink is a SubscriptionOpt for creating a subscription with a specific sink.
```

File: components/eventing-controller/testing/test_helpers.go [hunk 2]
```diff
@@ -329,11 +333,24 @@ func WithEventTypeFilter(s *eventingv1alpha1.Subscription) {
 	}
 }
 
-func WithValidSink(svcNs, svcName string, s *eventingv1alpha1.Subscription) {
-	s.Spec.Sink = GetValidSink(svcNs, svcName)
+// WithValidSink is SubscriptionOpt for creating a subscription with a valid sink that itself gets created from
```
Comment (FriedrichWilken) [inline components/eventing-controller/testing/test_helpers.go:336]:
```suggestion
// WithValidSink is a SubscriptionOpt for creating a subscription with a valid sink that itself gets created from
```

File: components/eventing-controller/testing/test_helpers.go [hunk 3]
```diff
@@ -304,13 +305,16 @@ func WithFilter(eventSource, eventType string) SubscriptionOpt {
...
+// WithEmptyEventTypeFilter is SubscriptionOpt for creating a subscription with an empty event type filter.
+//  Note that this is different from setting Filter to nil.
+func WithEmptyEventTypeFilter(subscription *eventingv1alpha1.Subscription) {
 	subscription.Spec.Filter = &eventingv1alpha1.BEBFilters{
 		Filters: []*eventingv1alpha1.BEBFilter{},
 	}
 }
 
-func WithEventTypeFilter(s *eventingv1alpha1.Subscription) {
+// WithDefaultEventTypeFilter is SubscriptionOpt for creating a subscription with the default event type filter.
```
Comment (FriedrichWilken) [inline components/eventing-controller/testing/test_helpers.go:316]:
```suggestion
// WithDefaultEventTypeFilter is a SubscriptionOpt for creating a subscription with the default event type filter.
```

File: components/eventing-controller/testing/test_helpers.go [hunk 4]
```diff
@@ -304,13 +305,16 @@ func WithFilter(eventSource, eventType string) SubscriptionOpt {
 	}
 }
 
-func WithEmptyFilter(subscription *eventingv1alpha1.Subscription) {
+// WithEmptyEventTypeFilter is SubscriptionOpt for creating a subscription with an empty event type filter.
```
Comment (FriedrichWilken) [inline components/eventing-controller/testing/test_helpers.go:308]:
```suggestion
// WithEmptyEventTypeFilter is a SubscriptionOpt for creating a subscription with an empty event type filter.
```

File: components/eventing-controller/testing/test_helpers.go [hunk 5]
```diff
@@ -252,7 +252,7 @@ func WithWebhookAuthForBEB(s *eventingv1alpha1.Subscription) {
 	}
 }
```
Comment (FriedrichWilken) [inline components/eventing-controller/testing/test_helpers.go:254]:
```suggestion

// WithWebhookForNATS is a SubscriptionOpt for creating a Subscription with a webhook set to the NATS protocol.
```

File: components/eventing-controller/pkg/subscriptionmanager/nats/nats_test.go [hunk 6]
```diff
@@ -35,7 +35,7 @@ func TestCleanup(t *testing.T) {
 	defer subscriber.Shutdown()
 
 	// Create test subscription
-	testSub := controllertesting.NewSubscription("test", "test", controllertesting.WithFakeSubscriptionStatus, controllertesting.WithEventTypeFilter)
+	testSub := controllertesting.NewSubscription("test", "test", controllertesting.WithFakeSubscriptionStatus, controllertesting.WithDefaultEventTypeFilter)
```
Comment (FriedrichWilken) [inline components/eventing-controller/pkg/subscriptionmanager/nats/nats_test.go:38]:
```suggestion
	testSub := controllertesting.NewSubscription("test", "test", 
		controllertesting.WithFakeSubscriptionStatus,
		controllertesting.WithDefaultEventTypeFilter,
	)
```

File: components/eventing-controller/pkg/subscriptionmanager/beb/beb_test.go [hunk 7]
```diff
@@ -32,7 +32,7 @@ func TestCleanup(t *testing.T) {
 
 	// create a Kyma subscription
 	subscription := controllertesting.NewSubscription("test", "test",
-		controllertesting.WithWebhookAuthForBEB, controllertesting.WithFakeSubscriptionStatus, controllertesting.WithEventTypeFilter)
+		controllertesting.WithWebhookAuthForBEB, controllertesting.WithFakeSubscriptionStatus, controllertesting.WithDefaultEventTypeFilter)
```
Comment (FriedrichWilken) [inline components/eventing-controller/pkg/subscriptionmanager/beb/beb_test.go:35]:
```suggestion
		controllertesting.WithWebhookAuthForBEB, 
		controllertesting.WithFakeSubscriptionStatus, 
		controllertesting.WithDefaultEventTypeFilter,
	)
```

**Comment by k15r** [comment]
Do we really want to remove the possibility to set a sinkvalidator? Currently we do not use it, i agree. But instead of removing it entirely we could also provide an option to change the sinkvalidator.

**Pull request closed without merging**